exec/java-exec/src/main/java/org/apache/drill/exec/server/options/InMemoryOptionManager.java

```diff
@@ -54,12 +56,32 @@ boolean setLocalOption(final OptionValue value) {
     return options.values();
   }
 
+  @Override
+  boolean deleteLocalOptions(final OptionType type) {
+    if (supportsOption(type)) {
+      options.clear();
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  boolean deleteLocalOption(final String name, final OptionType type) {
+    if (supportsOption(type)) {
+      options.remove(name);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
   /**
-   * Check to see if implementations of this manager support the given option value (e.g. check for option type).
+   * Check to see if implementations of this manager support the given option type.
    *
-   * @param value the option value
-   * @return true iff the option value is supported
+   * @param type option type
+   * @return true iff the type is supported
    */
-  abstract boolean supportsOption(OptionValue value);
+  abstract boolean supportsOption(OptionType type);
```

Comment:
you should probably rename this to `supportsOptionType` because we are effectively checking if the option type is supported. This will help the method be self explanatory